tests/v1/tpu/test_topk_topp_sampler.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-License-Identifier: Apache-2.0
+import math
+
+import torch
+
+from vllm.platforms import current_platform
+from vllm.v1.sample.ops.topk_topp_sampler import apply_top_k_top_p_tpu
+
+if current_platform.is_tpu():
+    import torch_xla.core.xla_model as xm
+
+DEVICE = xm.xla_device() if current_platform.is_tpu() else torch.device("cuda")
+
+BATCH_SIZE = 1024
+VOCAB_SIZE = 128 * 1024
+
+
+def test_topk_and_no_op_topp():
+    with torch.device(DEVICE):
+        if current_platform.is_tpu():
+            xm.set_rng_state(seed=33)
+        else:
+            torch.manual_seed(33)
+
+        logits = torch.rand((BATCH_SIZE, VOCAB_SIZE))
+
+        # Random top-k values between 1 and 9.
+        k = torch.randint(1, 10, (BATCH_SIZE, ))
+
+        # Set k=vocab_size for ~50% of requests in the batch (top-k disabled).
+        k.masked_fill_(torch.randint(0, 2, (BATCH_SIZE, ), dtype=bool),
+                       VOCAB_SIZE)
+
+        # Top-k only implementation
+        result1 = apply_top_k_top_p_tpu(logits=logits.clone(), k=k, p=None)
+
+        # Top-p + top-k
+        no_op_top_p = torch.tensor([1.0])
+        result2 = apply_top_k_top_p_tpu(logits=logits.clone(),
```

Comment:
nice test!

Reply:
Credit to @njhill's apply_top_k_only test!